### ISSUE TITLE
Process leaks file descriptors in the presence of errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2335]
|<img alt="eric.newton's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| eric.newton reported|
|-|-|
|Trac ID|trac#2335|
|Type|defect|
|Created|2006-12-27 16:30:17Z|

In twisted/internet/process.py, Twisted 2.4.0 on Linux w/2.6-smp Kernel.

multiple calls to os.pipe() are not protected against failures to allocate all the requested file descriptors, resulting in leakes that can bring down an entire server.

I get this when running as many processes in parallel as I possibly can.

Patch attached.

-Eric
ecn AT zenoss c0m

Attachments:

* [t](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5f9/5f95559ce118ad7a90ccf147c25ef96e06ace831/8efd86fb78a56a5145ed7739dcb00c78581c5375) (2578 bytes) - added by eric.newton on 2006-12-27 16:30:48Z - patch that cleans up file descriptors

<details><summary>Searchable metadata</summary>

```
trac-id__2335 2335
type__defect defect
reporter__eric_newton eric.newton
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__ 
time__1167237017000000 1167237017000000
changetime__1167237480000000 1167237480000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

This is a duplicate of [#2305](https://github.com/twisted/twisted/issues/2305).  Thanks for the patch, I'll make a note on the other ticket that it is here.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
